qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/AbstractRollingUpgradeTestCase.java

```diff
@@ -46,8 +47,8 @@ public String get() {
             .feature(FeatureFlag.TIME_SERIES_MODE);
 
         // Avoid triggering bogus assertion when serialized parsed mappings don't match with original mappings, because _source key is
-        // inconsistent
-        if (oldVersion.before(Version.fromString("8.18.0"))) {
+        // inconsistent. Assume "versionless" clusters (serverless) do not need this.
+        if (Version.tryParse(getOldClusterVersion()).map(v -> v.before(Version.fromString("8.18.0"))).orElse(false)) {
```

Comment:
I'm not totally clear on what this "assume" comment means. It looks like we might be doing something here that we have not confirmed is correct. If that's not the case, perhaps there's a way to reword it?